tests/migration.go

```diff
@@ -72,6 +79,92 @@ func ConfirmVMIPostMigration(virtClient kubecli.KubevirtClient, vmi *v1.VirtualM
 
 	By("Verifying the VMI's is in the running state")
 	Expect(vmi.Status.Phase).To(Equal(v1.Running), "the VMI must be in `Running` state after the migration")
+
+	return vmi
+}
+
+func setOrClearDedicatedMigrationNetwork(nad string, set bool) *v1.KubeVirt {
+	virtClient, err := kubecli.GetKubevirtClient()
+	Expect(err).ToNot(HaveOccurred())
+
+	kv := util.GetCurrentKv(virtClient)
+
+	// Saving the list of virt-handler pods prior to changing migration settings, see comment below.
+	listOptions := metav1.ListOptions{LabelSelector: v1.AppLabel + "=virt-handler"}
+	virtHandlerPods, err := virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(context.Background(), listOptions)
+	Expect(err).ToNot(HaveOccurred(), "Failed to list the virt-handler pods")
+
+	if set {
+		if kv.Spec.Configuration.MigrationConfiguration == nil {
+			kv.Spec.Configuration.MigrationConfiguration = &v1.MigrationConfiguration{}
+		}
+		kv.Spec.Configuration.MigrationConfiguration.Network = &nad
+	} else {
+		if kv.Spec.Configuration.MigrationConfiguration != nil {
+			kv.Spec.Configuration.MigrationConfiguration.Network = nil
+		}
+	}
+
+	res := UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
+
+	// By design, changing migration settings trigger a re-creation of the virt-handler pods, amongst other things.
+	//   However, even if SetDedicatedMigrationNetwork() calls UpdateKubeVirtConfigValueAndWait(), VMIs can still get scheduled on outdated virt-handler pods.
+	//   Waiting for all "old" virt-handlers to disappear ensures test VMIs will be created on updated virt-handler pods.
+	Eventually(func() bool {
+		for _, pod := range virtHandlerPods.Items {
+			_, err = virtClient.CoreV1().Pods(flags.KubeVirtInstallNamespace).Get(context.Background(), pod.Name, metav1.GetOptions{})
+			if err == nil {
+				return false
+			}
+		}
+		return true
+	}, 180*time.Second, 10*time.Second).Should(BeTrue(), "Some virt-handler pods survived the migration settings change")
+
+	// Ensure all virt-handlers are ready
```

Comment:
We are only checking they are running not ready. It would be better to check for readiness but we don't advertise the device plugin readiness.